plotlyJS

sglyon opened the pull request:
The goal of this PR is to replace some of the more verbose "by hand" figures being used in `dcc_graph` with ones produced by PlotlyJS.jl

Most of the change are in the user guide/tutorial section at the beginning of the docs. However, there are a couple examples in the dash_core_components section also.

Besides making those changes -- I cleaned up a bit of the language and found a handful of bugs in the code examples.

I also delete some Python and R files that I didn't believe belonged there -- but if I was mistaken on that please let me know and we will restore them!

sglyon committed:
update getting-started docs

sglyon committed:
remove RDatasets

sglyon committed:
update callbacks docs

sglyon committed:
update crossfilter docs for PlotlyJS

Remove R and py code

sglyon committed:
minor tweaks to sharing and FAQ pages

sglyon committed:
Update graph examples in dash_core_components

sglyon committed:
Update graphs in tabs example

sglyon committed:
update version of PlotlyBase used in docs

sglyon committed:
re-enable dash_vtk

mohdibntarek commented on DashDocs/src/chapters/basic_callbacks/examples/hello-slider.jl:
```diff
@@ -24,24 +23,21 @@ callback!(
             xaxis_type = "log",
-            xaxis_title = "GDP Per Capita",
-            yaxis_title = "Life Expectancy",
+            xaxis_title_text = "GDP Per Capita",
```
Was the old one wrong?

mohdibntarek commented on DashDocs/src/chapters/basic_callbacks/examples/multi-inputs.jl:
```diff
@@ -68,26 +70,20 @@ callback!(
+    return plot(
+        x, y,
         Layout(
-            xaxis_type = xaxis_type == "Linear" ? "linear" : "log",
```
Was this not functional?

mohdibntarek replied:
Ah I see, nvm that comment.

mohdibntarek commented on DashDocs/src/chapters/dash_core_components/Graph/examples/figure_graph.jl:
```diff
@@ -1,22 +0,0 @@
-using Dash, DashCoreComponents
```
Why remove this file?

sglyon replied:
Did not seem to be included anywhere. I can re-add if I was wrong

mohdibntarek replied:
Yes we need this file. It seems to be included too https://github.com/plotly/dash-julia-docs/pull/20/files#diff-0d967af92fd3f1cafcbd1d027fd2340b459bb625a5ec387337b1fa864713aba6R4.

mohdibntarek replied:
Please add it back then I am happy to merge this. Everything else looks good.

sglyon replied:
Thanks -- added back locally. Waiting on v0.1.6 to be released and will rebase this PR off that and push change

sglyon commented:
ref https://github.com/plotly/Dash.jl/issues/88

efmanu commented on Manifest.toml:
```diff
@@ -33,50 +45,62 @@ git-tree-sha1 = "ec77888ac3e78f9d372c2b533bdb52668f9e2b09"
-deps = ["CodecZlib", "DashBase", "DashCoreComponents", "DashHtmlComponents", "DashTable", "DataStructures", "HTTP", "JSON", "JSON2", "MD5", "PlotlyBase", "Sockets", "Test", "UUIDs"]
-git-tree-sha1 = "7c09315c7dbfd9510dd822fcee3910c4c139afd8"
+deps = ["CodecZlib", "DashBase", "DashCoreComponents", "DashHtmlComponents", "DashTable", "DataStructures", "HTTP", "JSON", "JSON2", "MD5", "PlotlyBase", "Requires", "Sockets", "Test", "UUIDs"]
+path = "/home/sglyon/.julia/dev/Dash"
```
Can you the actual Dash package in Julia registry instead of dev package?

sglyon replied:
Good catch -- will fix! Waiting on Dash.jl 0.1.6 to be released and will then make the change and push here

sglyon replied:
fixed!

efmanu commented on Manifest.toml:
```diff
@@ -98,9 +122,9 @@ version = "0.1.0"
 [[DashCoreComponents]]
 deps = ["DashBase"]
-git-tree-sha1 = "4f4935719119350b1f4c464341ef64743920ea74"
+path = "/home/sglyon/.julia/dev/DashCoreComponents"
```
Can you the actual DashCoreComponents package in Julia registry instead of dev package?

sglyon replied:
Good catch -- will fix! Waiting on Dash.jl 0.1.6 to be released and will then make the change and push here

sglyon replied:
fixed!

sglyon committed:
re-add figure_graph, update Dash to 0.1.6

sglyon committed:
change (x|y)axis_title_text => (x|y)axis_title

Needed downstream bug to be fixed in PlotlyBase -- https://github.com/sglyon/PlotlyBase.jl/commit/a82d8c3d13fbbbd5e911a805b0d40d6ca600c211

sglyon committed:
update plotlyjs version

sglyon committed:
Merge branch 'master' into plotlyJS

sglyon commented:
ok I think this is ready to go!
